### ISSUE TITLE
Fix derive missing docs build warnings

### ISSUE THREAD

**Issue by ascjones** [paritytech/jsonrpc#373]
Missing docs warnings for derive are due to missing docs in doctests: https://github.com/paritytech/jsonrpc/blob/master/derive/src/lib.rs#L12. @tomusdrw tried to fix this but the build failed https://ci.appveyor.com/project/paritytech/jsonrpc/builds/22025130

Also the whitespace needs fixing in the doctests.

**Comment by dvc94ch** [comment]
I think the warnings are due to `#[proc_macro_attribute]` expanding to
```rust
pub mod decls {
    extern crate proc_macro;
    #[rustc_proc_macro_decls]
    pub static _DECLS: &[proc_macro::bridge::client::ProcMacro] =
        &[proc_macro::bridge::client::ProcMacro::attr(
            "rpc",
            crate::rpc,
        )];
}
```
I don't really see a way to fix this other than removing `#[warn(missing_docs)]`.

**Comment by ascjones** [comment]
Makes sense, thanks @dvc94ch. Will keep this open in case anyone has an idea.
